File: city-tours-be/src/main/java/ivan/personal/tours/utility/utils/TypizedFile.java
package ivan.personal.tours.utility.utils;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TypizedFile {
    private byte[] file;
    private String contentType;
}

File: city-tours-fe/src/app/model/common/TypizedFile.ts
export class TypizedFile {

  file: string;
  contentType: string;

  constructor() {
  }

  public static toBase64Image(r: TypizedFile): string {
    return r != null && r.contentType != null ? "data:image/" + r.contentType + ";base64," + r.file : null;
  }
}
